src/Lower.cpp

```diff
@@ -62,32 +63,37 @@ using std::make_pair;
 
 Stmt lower(const vector<Function> &outputs, const string &pipeline_name, const Target &t, const vector<IRMutator *> &custom_passes) {
 
-    // Compute an environment
+    // Compute an wrapped_environment
     map<string, Function> env;
     for (Function f : outputs) {
         map<string, Function> more_funcs = find_transitive_calls(f);
         env.insert(more_funcs.begin(), more_funcs.end());
     }
 
+    // Create a deep-copy of the Func and substitute in wrapper funcs.
```

Comment:
More explicit comment: Create a deep-copy of the entire graph of Funcs, and substitute in wrapper Funcs